src/main/src/main/java/org/geoserver/security/urlchecks/URLCheck.java

```diff
@@ -0,0 +1,62 @@
+/* (c) 2023 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.security.urlchecks;
+
+import org.geotools.data.ows.URLChecker;
+
+/**
+ * Base class for configurable URL checks. It's {@link URLChecker}, but since it's configurable at
+ * runtime, it will be created/destroyed during the life of the application. Also, there will be
+ * multiple instances from the same check class. As a result, it's not well suited for direct SPI
+ * usage. {@link GeoserverURLChecker} is the stable checker plugged into SPI, which then looks up
+ * and uses implementations of this class hierarchy.
+ */
+public abstract class URLCheck implements URLChecker {
```

Comment:
Updating name.

The GeoServer-specific part of it is in its Javadoc, explaining why I had to split the configurable part out of SPI. It happens to implement URLChecker because it really needs the same methods, but it's accidental. A GeoTools implementation would not need getConfiguration(), either.